src/components/interface/ResolutionWarning.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  Button, Modal, Text, Title, Group, Stack,
+} from '@mantine/core';
+import { useEffect, useState } from 'react';
+import { IconAlertTriangle } from '@tabler/icons-react';
+import { useNavigate } from 'react-router';
+import { useStoreSelector } from '../../store/store';
+
+export function ResolutionWarning() {
+  const navigate = useNavigate();
+  const studyConfig = useStoreSelector((state) => state.config);
+
+  const minWidth = studyConfig.uiConfig.minWidthSize!;
+  const minHeight = studyConfig.uiConfig.minHeightSize!;
```

Comment:
You're forcing typescript to think these are defined, but maybe it's better for them to possibly be undefined? You'll have to update the logic in the resize handler, though
```suggestion
  const minWidth = studyConfig.uiConfig.minWidthSize;
  const minHeight = studyConfig.uiConfig.minHeightSize;
```